odoo/addons/base/models/res_users.py

```diff
@@ -1091,6 +1091,18 @@ def write(self, values):
             self._check_one_user_type()
         return res
 
+    # sudo helper methods to adjust implied groups
+    def _apply_group(self, implied_group):
+        superself = self.sudo()
+        if implied_group not in superself.implied_ids:
+            superself.implied_ids |= implied_group
+
+    def _remove_group(self, implied_group):
+        superself = self.sudo()
+        if implied_group in superself.implied_ids:
+            superself.implied_ids -= implied_group
+            implied_group.sudo().users -= superself.users
```

Comment:
Do not begin a method with `self=self.sudo()`, call it in sudo mode when you need it instead.

Reply:
ok changed